src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/OptionKey2.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using Roslyn.Utilities;
+
+#if CODE_STYLE
+using WorkspacesResources = Microsoft.CodeAnalysis.CodeStyleResources;
+#endif
+
+namespace Microsoft.CodeAnalysis.Options
+{
+    [NonDefaultable]
+    internal readonly struct OptionKey2 : IEquatable<OptionKey2>
```

Comment:
This is a clone of `OptionKey` as the code is very trivial. I have added implicit conversion to public type at the bottom of the file.